### PR TITLE
Adding scenario for restoring from a local backup

### PR DESCRIPTION
**Pull request by tyler-ball**
### Description

Copies the standalone-fresh-install scenario. But instead of running the pedant tests, restore an existing backup. This facilitates some testing we are trying to perform for the
`chef analyze` command.

### Issues Resolved

N/A

### Check List

- [x] New functionality includes tests
- [ ] All buildkite tests pass
- [ ] Full omnibus build and tests in buildkite pass
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>
- [x] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

File: terraform/aws/Makefile [hunk 1]
```diff
@@ -37,6 +41,7 @@ ifneq ($(INSTALL_VERSION),)
 	export TF_VAR_install_version_url ?= $(shell platform=$(TF_VAR_platform); for channel in unstable current stable; do mixlib-install download chef-server --url -c $$channel -a x86_64 -p $$(echo $${platform%-*} | sed 's/rhel/el/') -l $${platform\#\#*-} -v $(INSTALL_VERSION) 2>/dev/null && break; done | head -n1)
 endif
 
+export UPGRADE_VERSION ?= $(INSTALL_VERSION)
```
Comment (tyler-ball) [inline terraform/aws/Makefile]: I found defaulting this meant I didn't need to supply `UPGRADE_VERSION` when not running the upgrade scenario

File: terraform/aws/modules/aws_instance/amis.tf [hunk 2]
```diff
@@ -49,6 +49,23 @@ data "aws_ami" "rhel_8" {
   owners = ["309956199498"]
 }
 
+# identify ubuntu 14.04 ami
+data "aws_ami" "ubuntu_1404" {
```
Comment (tyler-ball) [inline terraform/aws/modules/aws_instance/amis.tf]: Customer is running old versions of Chef Server, which require older versions of Ubuntu, so I added this

File: terraform/aws/modules/aws_vpc/outputs.tf [hunk 3]
```diff
@@ -13,3 +13,7 @@ output "ipv4_cidr_block" {
 output "ipv6_cidr_block" {
   value = "${cidrsubnet(aws_vpc.default.ipv6_cidr_block, 8, 1)}"
 }
+
+output "region" {
```
Comment (tyler-ball) [inline terraform/aws/modules/aws_vpc/outputs.tf]: I found this output useful - feel free to delete it.

Comment (tyler-ball) [reply terraform/aws/modules/aws_vpc/outputs.tf]: I also get TF output running `make create-vpc` but not `make apply`. Am I doing something wrong?

File: terraform/aws/scenarios/omnibus-restore-backup/files/chef-server.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+opscode_erchef['keygen_start_size'] = 30
+
+opscode_erchef['keygen_cache_size']=60
+
+nginx['ssl_dhparam']='/etc/opscode/dhparam.pem'
+
+insecure_addon_compat false
+
+data_collector['token'] = 'foobar' unless data_collector.nil?
```
Comment (tyler-ball) [inline terraform/aws/scenarios/omnibus-restore-backup/files/chef-server.rb]: Copied this but added the `unless` clauses to ensure this config works with older server versions. Want me to copy this to the other `chef-server.rb` files in the other scenarios?

File: terraform/aws/scenarios/omnibus-restore-backup/main.tf [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    inline = [
+      "set -evx",
+      "echo -e '\nBEGIN INSTALL CHEF SERVER\n'",
+      "curl -vo /tmp/${replace(var.upgrade_version_url, "/^.*\\//", "")} ${var.upgrade_version_url}",
+      "sudo ${replace(var.upgrade_version_url, "rpm", "") != var.upgrade_version_url ? "rpm -U" : "dpkg -iEG"} /tmp/${replace(var.upgrade_version_url, "/^.*\\//", "")}",
+      "sudo chown root:root /tmp/chef-server.rb",
+      "sudo chown root:root /tmp/dhparam.pem",
+      "sudo mv /tmp/chef-server.rb /etc/opscode",
+      "sudo mv /tmp/dhparam.pem /etc/opscode",
+      "export CHEF_LICENSE=accept",
```
Comment (tyler-ball) [inline terraform/aws/scenarios/omnibus-restore-backup/main.tf]: Moving this from `--chef-license=accept` flag to an environment variable ensured it worked with an older Chef Server version which does not understand this flag

File: terraform/aws/Makefile (hunk 1 above)
Comment (PrajaktaPurohit) [reply terraform/aws/Makefile]: That is right!

File: terraform/aws/modules/aws_instance/amis.tf (hunk 2 above)
Comment (PrajaktaPurohit) [reply terraform/aws/modules/aws_instance/amis.tf]: Is this something we would need to run on a regular basis? Or it is only for this customer case?

File: terraform/aws/scenarios/omnibus-restore-backup/files/chef-server.rb (hunk 4 above)
Comment (PrajaktaPurohit) [reply terraform/aws/scenarios/omnibus-restore-backup/files/chef-server.rb]: That would be awesome!

**Comment by tyler-ball** [comment]
@christopher-snapp Any idea what to do about this license scout error? I'm not sure why it is failing on a Hab package - we don't distribute Hab in Chef Server, do we?

File: terraform/aws/modules/aws_instance/amis.tf (hunk 2 above)
Comment (tyler-ball) [reply terraform/aws/modules/aws_instance/amis.tf]: Since I am the first one to add it, it probably would not be needed on a regular basis. But is there a reason to not add it other than potential future maintenance (updating owner id, or search terms, etc.)?

File: terraform/aws/scenarios/omnibus-restore-backup/files/chef-server.rb (hunk 4 above)
Comment (tyler-ball) [reply terraform/aws/scenarios/omnibus-restore-backup/files/chef-server.rb]: Added in 8ed385c00

**Comment by tyler-ball** [comment]
Master is failing with the same license scout error

File: terraform/aws/Makefile (hunk 1 above)
Comment (christopher-snapp) [reply terraform/aws/Makefile]: Yeah, the variable names are confusing...especially if you aren't running an upgrade scenario.

File: terraform/aws/modules/aws_instance/amis.tf (hunk 2 above)
Comment (afiune) [reply terraform/aws/modules/aws_instance/amis.tf]: Let us not block this work to get merged since we would like other engineers to use it, if we need to add a test into the pipeline we should do it with a very dummy backup that doesn't need to be stored secretly (in vault)

File: terraform/aws/modules/aws_instance/amis.tf [hunk 6]
```diff
@@ -49,6 +49,23 @@ data "aws_ami" "rhel_8" {
...
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-trusty-14.04-amd64-server-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["099720109477"]
```
Comment (afiune) [inline terraform/aws/modules/aws_instance/amis.tf]: who is this? and would that work with other engineers? 🤔

File: terraform/aws/scenarios/omnibus-restore-backup/README.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+# Omnibus Standalone Fresh Install
```
Comment (afiune) [inline terraform/aws/scenarios/omnibus-restore-backup/README.md]: Is this the name we want to call this scenario? fresh install and restore maybe? or you can have a fresh install without restoring a backup?

**Comment by tyler-ball** [comment]
@christopher-snapp It can potentially be very small. Let me make a backup with something like 1 node, 1 cookbook, 1 env, etc.

File: terraform/aws/modules/aws_vpc/outputs.tf (hunk 3 above)
Comment (christopher-snapp) [reply terraform/aws/modules/aws_vpc/outputs.tf]: nah you aren't doing anything wrong

I'm sure there are a bazillion things that I should have done better.  My primary focus was trying to enable a developer to easily spin up ALL scenarios concurrently which changed what I thought should be output to the console for each scenario.

I'm totally open to suggestions and whatnot, but note that you have access to the buildkite pipeline version of this stuff which is even easier to launch the full matrix of tests without any setup of your workstation.

File: terraform/aws/modules/aws_instance/amis.tf (hunk 6 above)
Comment (christopher-snapp) [reply terraform/aws/modules/aws_instance/amis.tf]: That owner is Canonical (the company) so it'll work fine for discovering the AMI.

File: terraform/aws/scenarios/omnibus-restore-backup/README.md (hunk 7 above)
Comment (christopher-snapp) [reply terraform/aws/scenarios/omnibus-restore-backup/README.md]: Worth noting on the naming here is that I'll be removing the `omnibus-` prefix from all of the scenarios in a future refactoring commit as it is just a holdover from the previous implementation of testing.

**Comment by christopher-snapp** [comment]
> @christopher-snapp It can potentially be very small. Let me make a backup with something like 1 node, 1 cookbook, 1 env, etc. to see how large it is
> 
> construction

I certainly appreciate you adding this scenario.

If you were able to get some small backup (like you described) would there be any meaningful commands that could be run to verify that the `chef-server-ctl restore` went as planned (e.g. some knife command with a `grep -q` of the result?) ?  This would allow the test to fail out if something went wrong and then I could just toss it in the buildkite stack as another facet of the full test matrix.

**Comment by tyler-ball** [comment]
Just talked with @PrajaktaPurohit - she is going to file a ticket in the backlog to track the "run this scenario with a very small backup and verify that the backup loads correctly with `knife node list`" pipeline concern.

Our team's motivation for creating this scenario is having a quick way to create an ephemeral Chef Server with customer backups so we can test the `chef-analyze` tool. Because that tool is in very early development we do not need to hold up the Chef Server pipeline by adding verification for `chef-analyze` into the Chef Server pipeline. In the future when we add integration testing to `chef-analyze` we may decide to add it in that pipeline, or duplicate that pipeline and use this scenario in it, or something else. But we will address that at a future date.

File: terraform/aws/scenarios/omnibus-restore-backup/README.md (hunk 7 above)
Comment (tyler-ball) [reply terraform/aws/scenarios/omnibus-restore-backup/README.md]: I copied this from another `README.md` (guess which one) - changed the title to `Omnibus Restore Existing Backup`

**Comment by PrajaktaPurohit** [comment]
https://github.com/chef/chef-server/issues/1908

**Comment by PrajaktaPurohit** [comment]
The license-scout failure are unrelated to this PR. We are in the process of fixing those. But do not want to cause an holdup while that gets sorted.
